addons/account/models/account_move_line.py

```diff
@@ -3397,22 +3397,26 @@ def _get_child_lines(self, grouped=True):
                     treated_lines = []
                     for line in amls.sorted('sequence'):
                         if line.display_type == 'line_subsection' and not self.collapse_composition and line.collapse_composition:
-                            sub_section_lines = amls.filtered(lambda l: (l.parent_id == self or l.parent_id.parent_id == self) and l != self)
+                            sub_section_lines = amls.filtered(lambda l: (l.parent_id == self or l.parent_id.parent_id == self))
                             result.append({
                                 'name': line.name,
                                 'price_subtotal': sum(l.price_subtotal for l in sub_section_lines),
                                 'price_total': sum(l.price_total for l in sub_section_lines),
-                                'display_type': 'product',
+                                'display_type': line.display_type,
                                 'original_display_type': line.display_type,
                                 'quantity': 1,
                                 'discount': line.discount,
                             })
                             treated_lines.extend(sub_section_lines)
                         elif line not in treated_lines:
+                            subtotal = sum(l.price_subtotal for l in amls.filtered(lambda l: (l.parent_id == self or l.parent_id.parent_id == self)))
+                            total = sum(l.price_total for l in amls.filtered(lambda l: (l.parent_id == self or l.parent_id.parent_id == self)))
+                            if line.display_type in {'line_section', 'line_subsection'} and subtotal == 0:
+                                continue
```

Comment:
This seems to be fixing something but I wasn't able to figure it out from the commit message

also, should we use one of the float comparison functions instead of `== 0`